### PR TITLE
Fix issue #718. Editor freezes on unmatching bracket.

### PR DESCRIPTION
**Pull request by niilohlin**

### CODE REVIEW

**Comment by ethercrow** [comment]
Thank you for taking this! Looks like it fixes jumping forward from an unmatched '('. What happens when jumping backward from an unmatched ')'?

**Comment by niilohlin** [comment]
Dang, I did not think about that. I fixed it in commit https://github.com/niilohlin/yi/commit/8836670ca52d2fc80494f0868c19144b250dd264

File: src/library/Yi/Buffer/HighLevel.hs [hunk 1]
```diff
@@ -282,11 +282,16 @@ prevNParagraphs n = replicateM_ n $ moveB unitEmacsParagraph Backward
...
-  where go opened c | c == cStop && opened == 0 = return ()
-                    | c == cStop                = stepB >> readB >>= go (opened-1)
-                    | c == cStart               = stepB >> readB >>= go (opened+1)
-                    | otherwise                 = stepB >> readB >>= go opened
+goUnmatchedB dir cStart' cStop' = getLineAndCol >>= \position ->
+  stepB >> readB >>= go position (0::Int)
+  where go pos opened c
+           | c == cStop && opened == 0 = return ()
+           | c == cStop       = stepB >> readB >>= go pos (opened-1)
+           | c == cStart      = stepB >> readB >>= go pos (opened+1)
```
Comment (ethercrow) [inline src/library/Yi/Buffer/HighLevel.hs]: `c == cStart` branch doesn't check for EOF or SOF, wouldn't that fall into infinite loop?

Comment (niilohlin) [inline src/library/Yi/Buffer/HighLevel.hs]: I checked, that is correct. For example if you only insert ')' at the top of a document and then try to match it, you get a infinite loop. I'll fix it.

**Pull request merged**